core/src/main/scala/kafka/log/LogSegments.scala

```diff
@@ -185,7 +184,7 @@ class LogSegments(topicPartition: TopicPartition) {
    *         if it exists.
    */
   @threadsafe
-  def higherEntry(offset: Long): Option[Map.Entry[JLong, LogSegment]] = Option(segments.higherEntry(offset))
+  def higherEntry(offset: Long): Option[Map.Entry[Long, LogSegment]] = Option(segments.higherEntry(offset))
```

Comment:
Why do we use `Map.Entry` instead of tuple here? Is it to avoid the allocation?

Reply:
That's one reason. The other is that `Map.Entry` provided a better interface for attribute lookup via `Map.Entry.getKey()` and `Map.Entry.getValue()`, as compared to `._1` and `._2` in a tuple.